### PR TITLE
Development Final Code for Review

### PR DESCRIPTION
**Pull request by paulfrancis556**

### CODE REVIEW

File: CMP1903_A1_2324/Die.cs
Comment (Stampyswamp) [inline CMP1903_A1_2324/Die.cs:1]: +Correctly Laid Out
+Naming Convention follows brief
+Enough Comments to understand what is going on.

-Possibly (overkill) add a /// comment ontop of the function which explains what the function does.
#SUPER

File: CMP1903_A1_2324/Game.cs
Comment (Stampyswamp) [inline CMP1903_A1_2324/Game.cs:1]: +Plenty Of Comments
+All names are fitting (dice and die is a bit confusing but its understandable why these are the names)
+Code is efficient and Effective by using minimal lines.
#SUPER

File: CMP1903_A1_2324/Program.cs
Comment (Stampyswamp) [inline CMP1903_A1_2324/Program.cs:1]: +Nice and simple. No need for comment as you can easily understand what is happening.
-You could push this further and add an input system with defensive programming/ exception handling to be able to roll the dice more than one time, as per the briefs instructions.

File: CMP1903_A1_2324/Testing.cs
Comment (Stampyswamp) [inline CMP1903_A1_2324/Testing.cs:1]: +Fast and efficient.
+Plenty of Comments
-Again like before (Overkill) could use a /// style comment above the function to give a further summary
#SUPER

File: CMP1903_A1_2324/Die.cs
Comment (Stampyswamp) [reply CMP1903_A1_2324/Die.cs:1]: if you need help with writing a /// I would recommend the following:
https://learn.microsoft.com/en-us/visualstudio/ide/reference/generate-xml-documentation-comments?view=vs-2022

File: CMP1903_A1_2324/Game.cs
Comment (Stampyswamp) [reply CMP1903_A1_2324/Game.cs:1]: -Possibly explain the encapsulation with comments.

File: CMP1903_A1_2324/Program.cs
Comment (Stampyswamp) [reply CMP1903_A1_2324/Program.cs:1]: -If you do expand with a quick input system comments would be advised.

File: CMP1903_A1_2324/Testing.cs [hunk 1]
```diff
@@ -16,5 +18,24 @@ internal class Testing
...
+                Debug.Assert(valuesToTest.Item1[i] > 0 && valuesToTest.Item1[i] < 7, "Die value is out of bounds.");
+            }
+
+            //check the sum
+            Debug.Assert(valuesToTest.Item2 > 2 && valuesToTest.Item2 < 19, "Sum value is out of bounds.");
+
+        }
+
     }
 }
```
Comment (Stampyswamp) [inline CMP1903_A1_2324/Testing.cs:41]: Overall, Your code is very well thought out and written. this code is Ideal as it is compact and efficient. The code fits the brief as requested (-input system) and your code will report each roll individually which is an advanced functionality.

The use of comments are excellent as it makes the code very easy to read through and understand everything even while skimming through the text.

#All in All, Well Done. 😁

**Review by Stampyswamp — CHANGES_REQUESTED** [review]
Super Code.

File: CMP1903_A1_2324/Testing.cs (hunk 1 above)
Comment (Stampyswamp) [reply CMP1903_A1_2324/Testing.cs:41]: EXTRA- exception handling has been dealt with as there are currently no errors to arise.

File: CMP1903_A1_2324/Game.cs
Comment (ML-04) [inline CMP1903_A1_2324/Game.cs:1]: **Positive**
+ Not only do the comments provide a good explanation but there are many of them to provide what everything does.
+ The code is clear and concise.
+ The loop for is easy to understand.

**Negative**
- Provide a comment on the encapsulation method used.

File: CMP1903_A1_2324/Program.cs
Comment (ML-04) [inline CMP1903_A1_2324/Program.cs:1]: **Positive**
+ The code is clear and concise.

**Negative**
- Provide a comment explaining what this does.
- Expand on the current code to allow a user to roll the dice multiple times.

File: CMP1903_A1_2324/Testing.cs
Comment (ML-04) [inline CMP1903_A1_2324/Testing.cs:1]: **Positive**
+ The code is clear and concise.
+ Each piece of code is explained well with a comment.

**Review by paulfrancis556 — COMMENTED** [review]
Changes complete
XML comments added

**Pull request merged**